internal/resolve/catalog.go

```diff
@@ -170,6 +165,39 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 	return resolvedBundle, resolvedBundleVersion, priorDeprecation, nil
 }
 
+type resolutionErrorInfo struct {
+	PackageName     string
+	Version         string
+	Channels        []string
+	Catalogs        []string
+	InstalledBundle *ocv1alpha1.BundleMetadata
+	ResolvedBundles []*declcfg.Bundle
+}
+
+func buildResolutionError(rei resolutionErrorInfo) error {
```

Comment:
I tried this once (an error generation function) and there was opposition to it. At that time, it was called in multiple places, this seems to be simply pulling out the code into a function.

Reply:
(And of course, this requires changing all the error messages)

Reply:
I pulled it out into a separate function because the linter was complaining about the nesting depth being too high.

Regarding changing all the error messages, _I_ found it a bit more complex to read the previous error logic in a switch statement and thought it was a bit more readable using this new `strings.Builder` approach because it is more linear in nature (i.e as you read each line of the code you see the scenarios that result in new information being added to the error string we return).

Reply:
If folks prefer the previous approach, I can revert this change.

Reply:
I don't care, I just wanted to make the note.